src/Containers/App/App.js

```diff
@@ -8,37 +8,42 @@ import { fetchNotes } from '../../thunks/fetchNotes'
 import { Header } from '../../Components/Header/Header'
 
 class App extends Component {
+
   componentDidMount() {
     this.props.fetchNotes()
   }
 
   render() {
-    const { notes, loading } = this.props
+    const { notes, loading, history } = this.props
     return (
       <div>
         <Fragment>
           <Header/>
           {loading && <AlternateScreen text='Loading notes...' />}
-          <Switch>
-            <Route exact path="/" component={NotesSection} />
-            <Route exact path="/new-note" component={NotesSection} />
-            <Route exact path="/notes/:id" component={NotesSection} />
-            <Route exact path='/not-found' render={() => <AlternateScreen text='404 Page Not Found' />} />
-            <Route path='*' render={() => <Redirect to={'/not-found'}/>} />
-          </Switch>
-          <Route path="/new-note" component={NoteForm} />
-          <Route
-            path="/notes/:id"
-            render={({ match }) => {
-              const { id } = match.params
-              const note = notes.find((note) => note.id === id)
-              if (note) {
-                return <NoteForm {...note} />
-              } else {
-                return <Redirect to={'/not-found'} />
-              }
-            }}
-          />
+          {!loading && 
+            <div>
+              <Switch>
+                <Route exact path="/" component={NotesSection} />
+                <Route exact path="/new-note" component={NotesSection} />
+                <Route exact path="/notes/:id" component={NotesSection} />
+                <Route path='*' render={() => <AlternateScreen text='404 Page Not Found' />} />
+                <Route path='/not-found' render={() => <AlternateScreen text='404 Page Not Found' />} />
```

Comment:
What is the reasoning behind two not found routes and another /notes/:id route?

Reply:
The '/not-found' route corresponds to a redirect that happens when a user tries to go to a notes/:id page that doesn't exist.